functions/handler.py

```diff
@@ -4,60 +4,47 @@
 import os
 
 REGION = os.getenv('REGION', 'us-west-2')
-service = os.getenv('efServiceName', 'default')
+service = os.getenv('SERVICE', 'default')
 sqs = boto3.client('sqs', region_name=REGION)
 ssm = boto3.client('ssm', region_name=REGION)
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
 
 
 def lambda_handler(event, context):
-    # This parsing can and will be improved in another PR
-    if 'requestBody' in event['body']:
-        returnDict = dict()
-        returnDict['details'] = {}
-        event = json.loads(event['body'])
-        event_body = event['requestBody']
-        if 'event' not in event_body or 'payload' not in event_body:
-            return {
+    # print(json.dumps(event))
+    # This parsing can and will be improved in the future
+    logger.info("{} service initialized.".format(service))
+    aws_event = event
+    returnDict = dict()
+    returnDict['details'] = {}
+
+    # See if submitted POST body is a valid JSON
+    try:
+        request_body = json.loads(aws_event['body'])
+    except Exception as e:
+        logger.error("Cannot parse event: {}, context: {}, exception: ".format(str(request_body), context.function_name, e))
+        return {
                 'statusCode': 400,
                 'body': json.dumps('Bad Request')
-            }
+        }
 
-        returnDict['summary'] = event_body['event']
-        returnDict['source'] = 'api_aws_lambda'
-        returnDict['category'] = service
-        returnDict['eventsource'] = service + '_api'
-        returnDict['hostname'] = 'marketplace.'+ service + '.us'
-        returnDict['tags'] = service
-        returnDict['processname'] = service + 'connect'
-        returnDict['processid'] = 'none'
-        returnDict['severity'] = 'INFO'
+    returnDict['summary'] = '{}_event'.format(service)
+    returnDict['source'] = 'api_aws_lambda'
+    returnDict['category'] = service
+    returnDict['eventsource'] = service + '_api'
+    returnDict['hostname'] = '{}_host'.format(service)
+    returnDict['tags'] = service
+    returnDict['processname'] = service
+    returnDict['processid'] = 'none'
+    returnDict['severity'] = 'INFO'
+    returnDict['details'] = json.loads(aws_event['body'])
 
-        if 'payload' in event_body:
```

Comment:
Curious why we removed this, it only renames specific pieces of the event to the format ES can ingest. It does not restrict the sending to sqs if these don't exist.

Reply:
I think we should still detect if a payload is present, all webhooks should have this  field somewhere in the json. Unless you think this handler could be used for much more than just webhooks, curious as to your thoughts?

Reply:
There is a point in saying the handler could be used for taking events other than from webhooks, but if we have to do that we can make a different handler / function / endpoint to address that I think.

I will add checking the presence of a payload as well.